components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+        set description(value) {
+            this._description = value || '';
+        }
+
+        /** @return {array} gets the buttons */
+        get buttons() {
+            return this._buttons || [];
+        }
+
+        /** @param {array} value, set the buttons */
+        set buttons(value) {
+            this._buttons = value || [];
+            this.render();
+        }
+
+        /** @return {string} gets the persist value */
+        get persist() {
+            return this._persist;
+        }
+
+        /** @param {string} value, sets the persist value */
+        set persist(value) {
+            this._persist = (value === true);
+        }
+
+        /** @return {string} gets the mode value */
+        get mode() {
+            return this.getAttribute('mode') || 'edit';
+        }
+
+        /** @param {string} value, sets the mode value */
+        set mode(value) {
+            this.setAttribute('mode', value);
+        }
+
+        /** @return {booleam} gets the disable-validation value */
+        get disableValidation() {
+            return (this.getAttribute('disable-validation') === 'true');
+        }
+
+        /** @param {string} value, sets the disable-validation value */
+        set disableValidation(value) {
+            this.setAttribute('disable-validation', value === true);
+        }
+
+        /** @return {string} gets the remoteImageCapture value */
+        get remoteImageCapture() {
+            return this._remoteImageCapture;
+        }
+
+        /** @param {string} value, sets the remoteImageCapture value */
+        set remoteImageCapture(value) {
+            this._remoteImageCapture = value === 'true';
+        }
+
+        /** @return {string} gets thew reset value */
+        get reset() {
+            return this.getAttribute('reset');
+        }
+
+        /** @param {string} value, gets the reset value */
+        set reset(value) {
+            return this.setAttribute('reset', value);
+        }
+
+        /**
+         * @description Loads the schema
+         * @returns {undefined} nothing.
+        */
+        loadSchema() {
+
+            const schemaUrl = this.src;
+
+            return componentSupport.request({ url: schemaUrl, dataType: 'json' }).then(data => {
+                this._schema = [data];
+                this.render();
+
+                // fire on schema loaded event
+                this.onschemaloaded(this, data.title || '', data);
+
+                // apply session stored form data if it exists
+                if (this.persist && window.sessionStorage && sessionStorage[this.src]) {
+
+                    const formData = sessionStorage[this.src] || '';
+
+                    if (formData !== '') {
+                        this.populate(JSON.parse(formData));
+                    }
+                }
+            }).catch(err => {
+                // eslint-disable-next-line no-alert
+                alert(`Schema not found! ${err}`);
```

Comment:
I've removed the alert, this way we see the error in the console which is much more useful.